src/train.py

```diff
@@ -101,18 +135,37 @@ def main(argv):
             labels = [CLASS_INDEX[label] for label in labels]
             labels = torch.tensor(labels).to(DEVICE)
 
-            # Zero the parameter gradients
-            optimizer.zero_grad()
+            style_vector = torch.rand((BATCH_SIZE, 100), device=DEVICE)
+            char_vec = char_vector(BATCH_SIZE, DEVICE)
+
+            # Generate characters
+            charsGenerated = gen(char_vec, style_vector)
+
+            # Zero the parameter gradients for the discriminator
+            optimizerDisc.zero_grad()
+
+            # Train the discriminator
+            prediction = disc(font)
+            lossDisc = criterionDisc(prediction, labels)
```

Comment:
The discriminator should be updated using the generated as well, not just the real data.